### ISSUE TITLE
DirAccess.get_current_dir() returning inconsistent trailing / between macOS and Windows

### ISSUE THREAD

**Issue by lostminds** [godotengine/godot#90984]
### Tested versions

4.3dev4

### System information

macOS 14.4.x and Windows 11

### Issue description

[DirAccess.get_current_dir()](https://docs.godotengine.org/en/stable/classes/class_diraccess.html#class-diraccess-method-get-current-dir) returns the current directory path of a DirAccess object. However, it seems that on macOS it can return the path with a "/" at the end, while on windows it does not. 

This is an issue if you want to use this path to build other paths since you can't rely on it being the same between platforms.

### Steps to reproduce

1. Have a project with a subfolder in your res:// project folder. In this example "graphics".
2. Run the following code:
```
var dirAccessTest:DirAccess = DirAccess.open("res://graphics/")
print("dirAccessTest.get_current_dir(): "+dirAccessTest.get_current_dir())
```
On macOS this will print out 
`dirAccessTest.get_current_dir(): res://graphics/`
On windows it will print out
`dirAccessTest.get_current_dir(): res://graphics`

If you instead init the DirAccess with a path without trailing slash `dirAccessTest:DirAccess = DirAccess.open("res://graphics")` it works the same on both platforms (and still accesses the correct directory), so I'm guessing this is the intended and expected behavior and it's just a case of the macOS DirAccess implementation not properly removing any trailing slash it was initiated with for `get_current_dir()`

### Minimal reproduction project (MRP)

The script lines above should be sufficient to reproduce the issue.

**Comment by bruvzg** [comment]
> This is an issue if you want to use this path to build other paths since you can't rely on it being the same between platforms.

This should not be an issue if you are building a path with `String.path_join` (which is the intended way to build paths).

> macOS DirAccess implementation not properly removing any trailing slash

Most likely it's the issue with generic Unix implementation (should be also reproducible on Linux and Android), macOS specific DirAccess path is only handling drive list and path case sensitivity.

**Comment by lostminds** [comment]
> This should not be an issue if you are building a path with `String.path_join` (which is the intended way to build paths).

Right, I didn't know about `String.path_join` and that's probably a better way to do this I'll use going forward. But with my method of just adding strings together at least I found this inconsistency.

**Comment by timothyqiu** [comment]
> This should not be an issue if you are building a path with `String.path_join` (which is the intended way to build paths).

Probably not directly related to `DirAccess`, but I encoutered a similar issue while debugging the "Open in Terminal" feature of the editor. Some terminal emulators behave differently based on whether there is a trailing backslash.

A bit annoying having to check & strip trailing backslashes.
